---
title: "gbdr - genome browsing with dalliance in R"
author: "Florian Uhlitz"
date: '`r format(Sys.time(), "%d %B, %Y")`'
output: 
  html_document:
    theme: cosmo
    toc: yes
---

This is a sketch for a [htmlwidget](https://htmlwidgets.org) wrapper for JavScript genome browser [dalliance](https://biodalliance.org).

# Quick guide

A dalliance function call generates a genome browser htmlwidget with a given reference genome (eg. GRCh38) and a given gene annotation (eg. GENCODEv21):

```{r load_dalliance, results = "asis", echo = F, warnings = F, message = F}

source("R/dalliance.R")

```

# Example browser with tracks

```{r browser_w_tracks, results = "asis", echo = F, warnings = F, message = F}

dalliance(input = example)

```


# Workflow

A common workflow loads... 

* a public reference genome (via DAS)
* a public gene annotation (via DAS)
* custom tracks with
    + quantitative data: `.bigbed`, `.bigwig` or indexed `.bam.`
    + or qualitative data: eg. `.gff` and `.vcf` (future) 

Where grouped tracks should be added in lists. In general, syntax to build a genome browser could resemble the [leaflet](https://rstudio.github.io/leaflet/) syntax to build maps.


```{r eval = F}

dalliance %>% 
  addReference("GRCh38") %>% 
  addAnnotation("GENCODEv21") %>% 
  addTrack(list("my_track_A_rep1", "my_track_A_rep2"), 
            list("my_track_B_rep1", "my_track_B_rep2")) %>% 

```





